CRM/Contact/BAO/Contact.php

```diff
@@ -2248,7 +2244,9 @@ public static function formatProfileContactParams(
             $value,
             $type,
             $valueId,
-            $contactID
+            $contactID,
```

Comment:
In other words do not only format the fields the logged in user has permission to (it's hard to imaging the field formatting function would ever be the right place to apply an ACL... but in the merge situation it's causing data loss)